pdns/packethandler.cc

```diff
@@ -1241,7 +1241,7 @@ DNSPacket *PacketHandler::doQuestion(DNSPacket *p)
       return r;
     }
     
-    if(!B.getAuth(p, &sd, target)) {
+    if(!B.getAuth(target, &sd, (p->qtype == QType::DS))) {
       DLOG(L<<Logger::Error<<"We have no authority over zone '"<<target<<"'"<<endl);
```

Comment:
Please keep all logic inside getAuth(). If you pass qtype to getAuth() the code will become a lot more readable.

if(!B.getAuth(target, &sd, p->qtype) {  or if(!B.getAuth(target,  p->qtype, &sd) {
vs
if(!B.getAuth(target, &sd, (p->qtype == QType::DS))) {